todo/todo.py

```diff
@@ -183,344 +173,370 @@ async def showsettings(self, ctx):
-        if await self.config.user(ctx.author).detailed_pop():
-            if len(popped_todos) > 1:
-                popped_todos = "\n".join(popped_todos)
-                msg += f"Removed these todo reminders!\n{popped_todos}"
-            # Since this is gonna be `1` I don't have to check for it
-            elif len(popped_todos):
-                popped_todos = popped_todos[0]
-                msg += f"Removed this todo reminder!\n{popped_todos}"
-            if len(failed) > 1:
-                failed = "\n".join(failed)
-                msg += f"\nCould not remove todos at these indexes\n{failed}"
-            elif len(failed):
-                failed = failed[0]
-                msg += f"\nCould not remove a todo at this index\n{failed}"
+    @todo.command()
+    async def remove(self, ctx, *indexes: positive_int):
+        """Delete todos"""
+        if not indexes:
+            return await ctx.send_help(ctx.command)
+        todos = await self.config.user(ctx.author).todos()
+        indexes = [x - 1 for x in indexes]
+        if not todos:
+            return await ctx.send(_no_todo_message.format(prefix=ctx.clean_prefix))
         else:
-            if len(popped_todos) > 1:
-                msg += "Removed those todos!"
-            elif len(popped_todos):
-                msg += "Removed that todo!"
-            if len(failed):
-                msg += "\nFailed to remove some todos!"
-        if not msg:
-            msg = "Hm, something went wrong"
-        await ctx.send(msg)
-        await self._maybe_auto_sort(author=ctx.author)
+            removed, failed = 0, 0
+            async with ctx.typing():
+                async with self.config.user(ctx.author).todos() as todos:
+                    for index in indexes:
+                        try:
+                            todos.pop(index)
+                        except IndexError:
+                            failed += 1
+                        else:
+                            removed += 1
+            msg = "Done."
+            detailed = await self.config.user(ctx.author).detailed_pop()
+            if detailed:
+                if removed:
+                    msg += f"\nRemoved: {removed} {'todos' if removed > 1 else 'todo'}"
+                if failed:
+                    msg += f"\nFailed to remove: {failed} {'todos' if failed > 1 else 'todo'}"
+            await ctx.send(msg)
+            await self._maybe_autosort(ctx)
+
+    @todo.command(aliases=["delall"])
+    async def removeall(self, ctx):
+        """Remove all of your todos"""
+        await ctx.send(
+            "WARNING, this will remove **ALL** of your todos. Would you like to remove your todos? (y/n)"
+        )
+        pred = MessagePredicate.yes_or_no(ctx)
+        try:
+            await self.bot.wait_for("message", check=pred)
+        except asyncio.TimeoutError:
+            await ctx.send("Okay, I won't delete your todos")
+        else:
+            if pred.result is False:
```

Comment:
We can simplify this as `result` returns a bool.

```suggestion
            if not pred.result:
```